examples/advanced/demo_optimizing_physics_parameter.py

```diff
@@ -0,0 +1,268 @@
+resulting in a highly ill-posed inverse problem.
+
+In this example, we only focus on a much more simpler problem: given the measurement :math:`y` and the ground truth :math:`x`, find the parameter :math:`\theta`.
+This can be reformulated as the following optimization problem:
+
+.. math::
+    \min_{\theta} \frac{1}{2} \|A(x, \theta) - y \|^2
+
+This problem can be addressed by first-order optimization if we can compute the gradient of the above function with respect to :math:`\theta`.
+The dependence between the operator :math:`A` and the parameter :math:`\theta` can be complicated.
+Physics classes in DeepInverse are implemented in a differentiable (from a programming viewpoint) manner.
+We can leverage the automatic differentiation engine provided in Pytorch to compute the gradient of the above loss function w.r.t to the physics parameters :math:`\theta`.
+"""
+
+# %%
+# Import required packages
+#
+import deepinv as dinv
+import torch
+from tqdm import tqdm
+import matplotlib.pyplot as plt
+
+device = "cuda" if torch.cuda.is_available() else "cpu"
+dtype = torch.float32
+
+# %%
+# Define the physics
+# ------------------
+#
+# In this first example, we use the convolution operator, defined in the :class:`deepinv.physics.Blur` class.
+# We also generate a random convolution kernel of motion blur
+
+generator = dinv.physics.generator.MotionBlurGenerator(
+    psf_size=(25, 25), rng=torch.Generator(device), device=device
+)
+true_kernel = generator.step(1, seed=123)["filter"]
+physics = dinv.physics.Blur(noise_model=dinv.physics.GaussianNoise(0.02), device=device)
+
+# %% Load and example image and compute the measurement
+x = dinv.utils.load_url_image(
+    dinv.utils.demo.get_image_url("celeba_example.jpg"),
+    img_size=256,
+    resize_mode="resize",
+).to(device)
+
+y = physics(x, filter=true_kernel)
+
+dinv.utils.plot([x, y, true_kernel], titles=["Sharp", "Blurry", "True kernel"])
+
+# %%
+# Define an optimization algorithm
+# --------------------------------
+#
+# The convolution kernel lives in the simplex: positive entries summing to 1.
+# We can use one of the most basic optimization algorithm -- Projected Gradient Descent.
+# The following function allows one to compute the gradient of the loss function with respect to the convolution kernel and the orthogonal projection onto the simplex, by a sorting algorithm (Reference: Large-scale Multiclass Support Vector Machine Training via Euclidean Projection onto the Simplex
+# Mathieu Blondel, Akinori Fujino, and Naonori Ueda)
+#
+
+
+def loss_fn(physics, x, y, filter):
+    y_hat = physics.A(x, filter=filter)
+    loss = (y_hat - y).pow(2).mean()
+    return 0.5 * loss
+
+
+def gradient(physics, x, y, filter):
+    filter = filter.clone().requires_grad_(True)
+    with torch.enable_grad():
+        y_hat = physics.A(x, filter=filter)
+        loss = torch.nn.functional.mse_loss(y_hat, y)
```

Comment:
here I would suggest to use a data fidelity instead, ie define `data_fidelity = dinv.optim.L2()`
```suggestion
        physics.update(filter=filter)
        loss = data_fidelity(y=y, x=x, physics=physics) / y.numel()
```

Normalize by `y.numel()` if you want to have the same loss value as the current one